chromium_src/components/sync_device_info/device_info_sync_bridge_unittest.cc

```diff
@@ -84,5 +84,46 @@ TEST_F(DeviceInfoSyncBridgeTest, RemoteDelete) {
   EXPECT_THAT(ReadAllFromStore(), UnorderedElementsAre(Pair(kLocalGuid, _)));
 }
 
+TEST_F(DeviceInfoSyncBridgeTest, LocalDeleteWhenOffline) {
+  InitializeAndMergeInitialData(SyncMode::kFull);
+  ASSERT_EQ(1u, bridge()->GetAllDeviceInfo().size());
+  ASSERT_EQ(1, change_count());
+  ASSERT_FALSE(ReadAllFromStore().empty());
+
+  const DeviceInfoSpecifics specifics = CreateSpecifics(1, base::Time::Now());
+  auto error = bridge()->ApplySyncChanges(bridge()->CreateMetadataChangeList(),
+                                          EntityAddList({specifics}));
+
+  ASSERT_FALSE(error);
+  ASSERT_EQ(2u, bridge()->GetAllDeviceInfo().size());
+  ASSERT_EQ(2, change_count());
+
+  const std::string kLocalGuid = CacheGuidForSuffix(kLocalSuffix);
+  ON_CALL(*processor(), IsEntityUnsynced(kLocalGuid))
+      .WillByDefault(Return(true));
+  EXPECT_CALL(*processor(), Delete(kLocalGuid, _)).Times(1);
+
+  bool deleted_device_info_sent = false;
+  base::RunLoop loop;
+  bridge()->DeleteDeviceInfo(
```

Comment:
this test actually last for 6 seconds. Not sure, is that possible to make it instant.

Reply:
Thanks. Playing with `ScopedMockTimeMessageLoopTaskRunner` didn't gave a good result, but `scoped_mock_time_message_loop_task_runner.h` contains a comment 
```
// Note: Use TaskEnvironment + TimeSource::MOCK_TIME instead of this in unit
// tests.
```
It turned out `DeviceInfoSyncBridgeTest` already uses  `TaskEnvironment` so to avoid patches I have to make `BraveDeviceInfoSyncBridgeTest` with `TaskEnvironment(MOCK_TIME)` to run `LocalDeleteWhenOffline`

Reply:
how about `test::TaskEnvironment::FastForwardBy(base::TimeDelta::FromSeconds(6))`

Reply:
thanks, just tried, but this didn't help.
Comments near the `FastForwardBy` says
```
// Only valid for instances using TimeSource::MOCK_TIME.
```
and I got `DCHECK(mock_time_domain_);` fired. Original `task_environment` for `DeviceInfoSyncBridgeTest` is created without `MOCK_TIME` flag and I have to make a new `BraveDeviceInfoSyncBridgeTest` with `MOCK_TIME` - then it doesn't require to call `FastForwardBy.

Reply:
what about this?
```
namespace base {
namespace test {
namespace {

class TaskEnvironmentMockTime : public TaskEnvironment {
 public:
  TaskEnvironmentMockTime()
      : TaskEnvironment(
            IsMockTimedTest(
                testing::UnitTest::GetInstance()->current_test_info()->name())
                ? TimeSource::MOCK_TIME
                : TimeSource::DEFAULT) {}

  static bool IsMockTimedTest(base::StringPiece test_name) {
    return test_name == "LocalDeleteWhenOffline";
  }
};

}  // namespace
}  // namespace test
}  // namespace base

#define TaskEnvironment TaskEnvironmentMockTime
```

Reply:
@goodov this is cool and this works :+1:

Reply:
pushed fix, thanks @goodov